expose-utility-functions-to-states.md

```diff
@@ -0,0 +1,133 @@
+- Feature Name: Expose utility functions to states, but not to the CLI or template context
+- Start Date: 2023-06-06
+- SEP Status: Draft
+- SEP PR: https://github.com/saltstack/salt-enhancement-proposals/pull/70
+- Salt Issue: https://github.com/saltstack/salt/pull/64287#pullrequestreview-1451646261
+
+# Summary
+[summary]: #summary
+
+Provide a means for the loader to access utility functions, but keep them from
+being added to the CLI or template context.
+
+# Motivation
+[motivation]: #motivation
+
+It is a policy of the Salt core team not to add any new utility functions to
+the `__salt__` loader (thus exposing them to the Salt CLI and the template
+context). The core team's proposed alternative, however, is to do an OS check
+in the state module, then a late import of the salt execution module, and
+finally to invoke the utility function directly. While this option technically
+works, it is unequivocally a bad idea, with several complications:
+
+1. The state modules were never supposed to include platform-specific code. I
+   was even [advised to
+   remove](https://github.com/saltstack/salt/pull/3019#issuecomment-11680999)
+   grains checks from a state module some 10 years ago.
+
+2. Doing an OS check in a state module duplicates work already done in the
+   execution module's `__virtual__` function. If the conditions in that
+   `__virtual__` function are changed at a later date, all ad-hoc duplications
+   of this logic in state modules must be located and updated. It is only a
+   matter of when, not if, this will cause bugs to arise.
```

Comment:
There are other ways to maintain consistency on `__virtual__` checks.

Reply:
Then name them?